h2o-py/h2o/schemas/error.py

```diff
@@ -6,153 +6,51 @@
 # noinspection PyUnresolvedReferences
 from h2o.utils.compatibility import *  # NOQA
 
+from functools import partial
+from h2o.schemas import _ignored_schema_keys
+
 
 class H2OErrorV3(object):
-    """
-    """
-    def __init__(self, keyvals):
+    
+    def __init__(self):
         self._props = {}
-        self._endpoint = None
-        self._payload = None
+        self.endpoint = None
+        self.payload = None
+        
+    @classmethod
+    def make(cls, keyvals):
+        err = cls()
         for k, v in keyvals:
-            if k == "__meta" or k == "_exclude_fields" or k == "__schema": continue
-            if k in _h2oerror_v3_valid_keys:
-                if k.endswith("msg"): v = v.replace("ERROR MESSAGE:", "").strip()
-                self._props[k] = v
-            else:
-                raise AttributeError("Attribute %s cannot be set on H2OErrorV3 (= %r)" % (k, v))
-
-    @property
-    def stacktrace(self):
-        return self._props["stacktrace"]
-
-    @property
-    def timestamp(self):
-        return self._props["timestamp"]
-
-    @property
-    def error_url(self):
-        return self._props["error_url"]
-
-    @property
-    def exception_type(self):
-        return self._props["exception_type"]
-
-    @property
-    def exception_msg(self):
-        return self._props["exception_msg"]
-
-    @property
-    def dev_msg(self):
-        return self._props["dev_msg"]
-
-    @property
-    def http_status(self):
-        return self._props["http_status"]
-
-    @property
-    def msg(self):
-        return self._props["msg"]
-
-    @property
-    def values(self):
-        return self._props["values"]
-
-
-    @property
-    def endpoint(self):
-        return self._endpoint
-
-    @endpoint.setter
-    def endpoint(self, value):
-        self._endpoint = value
-
-    @property
-    def payload(self):
-        return self._payload
-
-    @payload.setter
-    def payload(self, value):
-        self._payload = value
-
-    def __repr__(self):
+            if k in _ignored_schema_keys: continue
+            if not hasattr(cls, k):
+                # we can add properties dynamically here as they are defined statically on the backend
+                # as properties of water.api.schemas3.H2OErrorV3 (or subclasses)
+                setattr(cls, k, property(partial(cls.__getitem__, name=k)))
```

Comment:
to avoid having to raise an unnecessary error when there is a version mismatch between PY client and backend, the properties are now generated on the fly on the first call, adapting dynamically to the backend schema.
This also avoid developpers to update this code when the schema is modified.